### ISSUE TITLE
Segfault on PocketBook Inkpad X

### ISSUE THREAD

**Issue by KucharczykL** [koreader/koreader#5887]
* KOReader version: v2020.02-12-g611c423_2020-02-18
* Device: PocketBook InkPad X

#### Issue

After installation the application didn't launch. I have added a new device entry to the `device` file but then it crashes with a different error message. Before it crashes, two message boxes are displayed on top of each other. One saying which file is being loaded, the one underneath says something about color and memory usage.

#### Steps to reproduce

Install on InkPad X and add these entries to the `koreader\frontend\device\pocketbook` file:

```
-- PocketBook Inkpad X
local PocketBook1040 = PocketBook:new{
    model = "PB1040",
    isTouchDevice = yes,
    hasKeys = yes,
    hasFrontlight = yes,
    hasColorScreen = yes,
    isAlwaysPortrait = yes,
    display_dpi = 227,
    emu_events_dev = "/var/dev/shm/emu_events",
}
```
and
```
elseif codename == "PB1040" then
    return PocketBook1040
```

Then open the application via the bottom slide-up menu.

##### `crash.log` (if applicable)
[crash.log](https://github.com/koreader/koreader/files/4230689/crash.log)

**Comment by NiLuJe** [comment]
Not actually a color screen, so, that can go away at the very least ^^.

**Comment by NiLuJe** [comment]
It might also not need `isAlwaysPortrait` either, but that's for once it no longer faceplants on startup, since it'll affect touch input.

**Comment by NiLuJe** [comment]
Other than that, it, err, eerily sounds like InkView is once again doign some nasty crap behind our back.

Ideally, a core dump and/or a strace would be mightily helpful, like it was in #5861...

Also, that appears to be a new FW release, so I wouldn't mind having a look at a copy of your `/usr/lib/libinkview.so` to double-check that theory...

**Comment by KucharczykL** [comment]
Just removing the color and portrait options didn't solve it. As for your last comment, I don't know how to do it. I tried to search for an answer but in vain. Could you give me some advice?

**Comment by Frenzie** [comment]
As for strace, I understand you might be able to run something like `strace -o log.txt -f koreader.app` on the device itself without any mods. Not sure though.

As for `/usr/lib/libinkview.so`, that should be a file on the device itself.

**Comment by NiLuJe** [comment]
You'll need shell access to the device for all of those things. There ought to be some somehwta current info about that buried in the PocketBook forum on MobileRead, but since I've never held a PB device myself, I can't really say much more.

**Comment by KucharczykL** [comment]
Thanks for the pointers! Managed to get root ssh access. These are the spoils of my endeavor:

[log.txt](https://github.com/koreader/koreader/files/4235144/log.txt)

[libinkview.zip](https://github.com/koreader/koreader/files/4235151/libinkview.zip)

**Comment by NiLuJe** [comment]
Ah.

Good news: No new InkView shenanigans.

Best news: You're using a build affected by #5861 (for.. some reason? Did you move an existing installation?).

TL;DR: A *clean* install from >= 2020.02-11 should do the trick ;).

(Well, with the new device block ^^).

**Comment by KucharczykL** [comment]
You are right. During testing different builds I probably forgot to replace the .app file as well. It works now! Thank you for your help.
